codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -557,8 +530,11 @@ private MqttSubscribePayload decodeSubscribePayload(
         final List<MqttTopicSubscription> subscribeTopics = new ArrayList<MqttTopicSubscription>();
         int numberOfBytesConsumed = 0;
         while (numberOfBytesConsumed < bytesRemainingInVariablePart) {
-            final Result<String> decodedTopicName = decodeString(buffer);
-            numberOfBytesConsumed += decodedTopicName.numberOfBytesConsumed;
+            int topicNameSize = decodeMsbLsb(buffer);
+            String decodedTopicName = buffer.toString(buffer.readerIndex(), topicNameSize, CharsetUtil.UTF_8);
+            buffer.skipBytes(topicNameSize);
```

Comment:
@normanmaurer would be nice to have `toString(int length, Charset charset)`, similar to `readCharSequence`

```
    @Override
    public CharSequence readCharSequence(int length, Charset charset) {
        CharSequence sequence = getCharSequence(readerIndex, length, charset);
        readerIndex += length;
        return sequence;
    }
```

that uses `readerIndex` as `index` and increments readerIndex.

Reply:
or even `toUtf8String(int length)` to avoid character check under the hood...

Reply:
> @normanmaurer would be nice to have `toString(int length, Charset charset)`, similar to `readCharSequence`
> 
> ```
>     @Override
>     public CharSequence readCharSequence(int length, Charset charset) {
>         CharSequence sequence = getCharSequence(readerIndex, length, charset);
>         readerIndex += length;
>         return sequence;
>     }
> ```
> 
> that uses `readerIndex` as `index` and increments readerIndex. So we can avoid 2 polymorphic method calls.

yeah might be worthwhile.